launch/launch/launcher.py

```diff
@@ -320,6 +391,9 @@ def __init__(self, launcher):
         super(AsynchronousLauncher, self).__init__()
         self.launcher = launcher
 
+    def terminate(self):
+        self.launcher.interrupt_launch()
```

Comment:
From terminate I would expect that it also waits until the thread has actually finished (`Thread.join()`?).